### PR TITLE
course run edit and change ownership

### PR DESCRIPTION
**Pull request by tasawernawaz**
[ECOM-7301](https://openedx.atlassian.net/browse/ECOM-7301)

- [x] change ownership on course run edit
- [x] warning messegs

### CODE REVIEW

File: course_discovery/apps/publisher/views.py [hunk 1]
```diff
@@ -174,6 +175,18 @@ def get_context_data(self, **kwargs):
         context['publisher_approval_widget_feature'] = waffle.switch_is_active('publisher_approval_widget_feature')
         context['publish_state_name'] = CourseRunStateChoices.Published
 
+        if has_role_for_course(course_run.course, user):
```
Comment (waheedahmed) [inline course_discovery/apps/publisher/views.py]: You can reduce the query count but changing it to `if context['can_edit']:`.

File: course_discovery/templates/publisher/course_run_detail/_edit_warning.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+<div id="editCourseRun" class="modal">
+    <div class="modal-content">
+        <h2 class="hd-2 emphasized">
+            <span class="icon fa fa-exclamation-triangle" aria-hidden="true"></span> {% trans "CAUTION" %}
+        </h2>
+        <span class="sr-only">CAUTION</span>
+        <p>
+            {% blocktrans trimmed with owner=object.course_run_state.owner_role reviewer_user='tasawer'%}
+                The {{ current_team_name }} is currently reviewing this course run. If you edit course run information, you might overwrite the team’s changes, and you will have to send the course to the {{ team_name }} for review again.
```
Comment (waheedahmed) [inline course_discovery/templates/publisher/course_run_detail/_edit_warning.html]: ..., and you will have to send the course run to ...

**Review by waheedahmed — APPROVED** [review]
minor comments, 👍 once addressed.

File: course_discovery/apps/publisher/forms.py [hunk 3]
```diff
@@ -147,8 +148,11 @@ def __init__(self, *args, **kwargs):
         user = kwargs.pop('user', None)
         edit_mode = kwargs.pop('edit_mode', None)
         organization = kwargs.pop('organization', None)
+        org_extension = None
         if organization:
             org_extension = OrganizationExtension.objects.get(organization=organization)
```
Comment (awais786) [inline course_discovery/apps/publisher/forms.py]: organization.organization_extension

Comment (tasawernawaz) [reply course_discovery/apps/publisher/forms.py]: @awais786 here `organization` is a str value id. e.g `1`, so we can't get `organization_extension` directly.

Comment (waheedahmed) [reply course_discovery/apps/publisher/forms.py]: Organization is not an object here, it will be a string.

Comment (waheedahmed) [reply course_discovery/apps/publisher/forms.py]: @tasawernawaz Your code may not work, you should use `organization_id=organization`.

Comment (tasawernawaz) [reply course_discovery/apps/publisher/forms.py]: @waheedahmed I did not changed this line in this PR, it was pre written and was working fine.

Comment (waheedahmed) [reply course_discovery/apps/publisher/forms.py]: Yeah, I just verified it and its working fine in both ways.

**Review by awais786 — APPROVED** [review]

File: course_discovery/apps/publisher/utils.py [hunk 4]
```diff
@@ -107,4 +107,5 @@ def has_role_for_course(course, user):
     Returns:
         bool: True, if user has a role for course; otherwise, False.
     """
+
```
Comment (waheedahmed) [inline course_discovery/apps/publisher/utils.py]: Remove this line.

**Pull request merged**